src/main/java/net/mcreator/ui/blockly/BlocklyJavascriptBridge.java

```diff
@@ -139,6 +140,16 @@ public class BlocklyJavascriptBridge {
 			case "spawnableEntity" -> openDataListEntrySelector(
 					w -> ElementUtil.loadAllSpawnableEntities(w).stream().filter(e -> e.isSupportedInWorkspace(w))
 							.toList(), "entity");
+			case "customEntity" -> openDataListEntrySelector(ElementUtil::loadCustomEntities, "entity");
+			case "entitydata_logic" -> openStringEntrySelector(
+					w -> ElementUtil.loadEntityDataListFromCustomEntity(w, customEntryProviders,
+							PropertyData.LogicType.class).toArray(String[]::new), "entity");
+			case "entitydata_number" -> openStringEntrySelector(

```

Comment:
Given this is in fact integer type, this should likely be called such across the system (machine names and in UI) to not confuse things.